internal/tiltfile/tiltfile_state.go

```diff
@@ -616,6 +616,41 @@ func (s *tiltfileState) assembleK8sV2() error {
 				r.name = opts.newName
 				s.k8sByName[r.name] = r
 			}
+
+			selectors := make([]k8sObjectSelector, len(opts.objects))
+			for i, o := range opts.objects {
+				s, err := selectorFromString(o)
+				if err != nil {
+					return errors.Wrapf(err, "Error making selector from string %s", o)
+				}
+				selectors[i] = s
+			}
+
+			for i, selector := range selectors {
+				if len(s.k8sUnresourced) == 0 {
+					return fmt.Errorf("Unable to find any unresourced matches for selector %s. All YAML has already been resourced", opts.objects[i])
+				}
+				matches := []k8s.K8sEntity{}
+				for _, e := range s.k8sUnresourced {
```

Comment:
IIUC, we also need to match against resourced objects or else this will behave confusingly.

e.g.:

we have foo:namespace and foo:service

resource1 adds the object "foo:namespace". there's only one match, so this is fine.
resource2 adds the object "foo". because "foo:namespace" is no longer uncategorized, this succeeds.

1. Per the design, this should fail because "foo" is ambiguous.
2. The result depends on whether we evaluate resource1 or resource2 first (if we evaluate resource2 first, it'll fail because "foo" will still be ambiguous), which is theoretically nondeterministic.

Similarly, if we have a foo:deployment and foo:secret and then k8s_resource('foo', objects=['foo']), it'll work, because foo:deployment isn't unresourced.

It seems easier to describe if we make the checks independent of the current state of resourcing.

Reply:
Yeah good call.